glutton.go

```diff
@@ -186,6 +198,8 @@ func (g *Glutton) registerHandlers() {
 // Shutdown the packet processor
 func (g *Glutton) Shutdown() (err error) {
 	defer g.logger.Sync()
+	g.cancel() // close all connection
+	time.Sleep(4 * time.Second)
```

Comment:
May be there exist a better but I am unable to find. The only link we have between program and goroutines is context. `cancel()` function signal  routines to abandon their work and does not wait for the work to stop. And in any case if fails then there will be definitely a goroutine leak. May be it is possible in future when we have connection counter so we can keep that counter synchronized with number of goroutines (connections) with help of context and on shutdown we wait until counter goes to zero.